zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/RoutingStateInitializer.java

```diff
@@ -20,29 +26,39 @@
  */
 public class RoutingStateInitializer implements ClusterConfigurationModifier {
 
+  private static final Logger LOG = LoggerFactory.getLogger(RoutingStateInitializer.class);
   private final boolean enablePartitionScaling;
-  private final int staticPartitionCount;
 
-  public RoutingStateInitializer(
-      final boolean enablePartitionScaling, final int staticPartitionCount) {
+  public RoutingStateInitializer(final boolean enablePartitionScaling) {
     this.enablePartitionScaling = enablePartitionScaling;
-    this.staticPartitionCount = staticPartitionCount;
   }
 
   @Override
   public ActorFuture<ClusterConfiguration> modify(final ClusterConfiguration configuration) {
-    if (configuration.routingState().isPresent() || !enablePartitionScaling) {
+    if (configuration.routingState().isPresent()) {
+      LOG.debug(
+          "Routing state is already initialized, skipping initialization: {}",
+          configuration.routingState());
       return CompletableActorFuture.completed(configuration);
     }
-
-    final var routingState = RoutingState.initializeWithPartitionCount(staticPartitionCount);
+    if (!enablePartitionScaling) {
+      LOG.debug("Partition scaling is not enabled, skipping initialization.");
+      return CompletableActorFuture.completed(configuration);
+    }
+    final var coordinator = MemberId.from("0");
+    final List<ClusterConfigurationChangeOperation> operations =
+        List.of(new UpdateRoutingState(coordinator, Optional.empty()));
+    var pendingChanges = configuration.pendingChanges();
+    if (pendingChanges.isEmpty()) {
+      pendingChanges = Optional.of(ClusterChangePlan.init(1L, operations));
+    }
```

Comment:
:question: This will be executed on all broker. That means all brokers are updating the `ClusterChangePlan`. This is anti-pattern. The correctness of the ClusterConfiguration is guaranteed under the assumption that only Coordinator can update "global" properties and each member can update their own properties.

Also I'm not sure whether you really need this operation to initialize routing state. This modifere is required only when upgrading to the first version where RoutingState is introduced. So it can be safely initialized from the static partition count.

If the goal was to make it also work after restore, why not add the operation during restore?

Reply:
Would just adding the initializer in the coordinator be enough to avoid conflicting updates?

Reply:
> Would just adding the initializer in the coordinator be enough to avoid conflicting updates?

I think adding an operation in initializer has further problems. If there is an ongoing cluster change plan in progress, this can interfere with it.  `ClusterConfigurationModifier` is a side track way which should be only used in special cases - for example, we added a new field to the `ClusterConfiguration`. So we need a way to add it to an existing `toology.meta`.